### DISCUSSION TITLE
Solar Spreadsheet to track production/usage, cost/payback, and other misc.

### DISCUSSION THREAD

**Discussion by hulkster** [jasonacox/Powerwall-Dashboard#545] [General]
I suspect many of you have your own spreadsheets to track production/usage ... so thought I would share mine.

I originally did this to calculate cost/payback (see the last tab) and thought it would be pretty simple spreadsheet, even after my system was installed and I started adding production/usage data from Powerwall & Xcel Energy.

But as often the case, it's gotten bigger and more complicated as I've added more data (such as heating/cooling days, PVWatts predictions, etc.) plus tried to address some of the conditions/situations of interest ... such as what happens if NEM changes. Note that the data granularity is by month ... so doing TOU pricing (especially if it's minute-by-minute) is not handled. 

At this point, it's mostly just monthly updates as I just plug in the latest set data and everything gets updated plus the spiffy graphs which allow you to look at a glance to see trends.

I tried to write the spreadsheet "generically" - i.e. if you spend some time to understand how it is laid out and organized, you should be able to easily punch in your data. 

I'd welcome any comments on how to make it better - thanks!
FYI that it is best viewed on a 4K monitor.

[2024_11_13_Solar_Analysis.xlsx](https://github.com/user-attachments/files/17736585/2024_11_13_Solar_Analysis.xlsx)

**Comment by jasonacox** [comment]
Nice @hulkster ! 

I do find it interesting to see the trend toward more efficiency.

<img width="675" alt="image" src="https://github.com/user-attachments/assets/c4b3c707-d690-4900-839c-eb69cdcf2a04">

**Reply by hulkster** [reply]
There's an explanation for that as I had a very extensive holiday lights display ([Christmas](https://www.komar.org/christmas/) & [Halloween](https://www.komar.org/halloween/)) display for many years ... but [_"hung up the Santa Hat"_ after 2014.](https://www.komar.org/christmas/2014/) 

So there was an expected electrical usage drop after that and it was somewhat constant from 2015 onward (not sure what happened in 2018 (really hot summer?) ... but I did replace our A/C unit in 2019) ... although has gone down more the last couple of years as my kids no longer live at home.

Because I have more data from 2021 onward, I can tell that 2023 was especially low because we had a really cool summer so much less A/C was used. You can see this on tabs for "Elec & Gas Data" (I color highlight abnormally warm/cold months based on heating/cooling days data from Xcel Energy) and also on the Utility Graphs - see below. For 2024, I'll probably be just below 6,500kWh.

I did swap out gas cooktop for induction in August/2024, so I expect that to cause a slight uptick in electrical usage ... and obviously a reduction in gas usage.

We actually looked at EV's recently (since I'm generating a surplus of power) but our 20-year old Minivan continues to run well (and we don't drive much) ... so why replace it.

But yes, reducing usage is a "good" thing ... the best form of conservation (and to save money) is NEGA-Watt!    ;-)

![2021_2024-10_Utility](https://github.com/user-attachments/assets/ed7da4bb-9421-41e5-ad0b-ed07935eae9a)

**Reply by jasonacox** [reply]
> ["hung up the Santa Hat" after 2014.](https://www.komar.org/christmas/2014/)

Oh wow... that's amazing @hulkster... Thanks for sharing! That's an incredible story. My uncle had celiac and often ended up in the hospital due to cross contamination. My daughter and I inherited it as well, but not at severe and only diagnosed as severe allergy.

**Reply by hulkster** [reply]
Thanks for the nice words @jasonacox  - lots of fun stuff (and raising awareness/$$$ for a good cause) for a decade ... [but the real comedy was the initial three years before that when technology wasn't quite available to do it for real ...  ;-)](https://www.komar.org/christmas/hoax/)

Hope your Uncle is doing well - ditto you and your daughter - it's a PIA having to always be careful what you eat.

**Reply by hulkster** [reply]
I'm going to bump this thread because I've updated things after my first full calendar year of solar - 16 months total - WOOT!

It's been a lot of fun geeking out with the data - thanks @jasonacox and others for an excellent piece of software.

[Full writeup with pretty pictures here](https://www.komar.org/how-to/2023-tesla-solar-panels-house/) ... that includes links to the upgraded spreadsheet and my JSON dashboard.

I'd welcome any comments/suggestions on how to improve that plus anything else on the website.

**Reply by clukas1967** [reply]
+1 on installing critter guards all the way around the panels.  I did not do that at first, and after 2 years discovered all the pigeon poop had filled my rear gutter past overflowing (which I can't reach anymore) and started a forest almost 2 meters wide.

After I put in critter guards, now my neighbors are having to deal with it instead of me.
